store/tikv/mock-tikv/cop_handler_dag.go

```diff
@@ -220,10 +234,54 @@ func (h *rpcHandler) buildTopN(ctx *dagContext, executor *tipb.Executor) (*topNE
 		},
 	}
 
+	conds, err := convertToExprs(ctx.eval.StatementCtx, colIDs, pbConds)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	return &topNExec{
-		TopN:   topN,
-		eval:   ctx.eval,
-		heap:   heap,
-		colIDs: colIDs,
+		heap:       heap,
+		evalCtx:    ctx.evalCtx,
+		colIDs:     colIDs,
+		conditions: conds,
+		row:        make([]types.Datum, len(ctx.eval.ColumnInfos)),
 	}, nil
 }
+
+type evalContext struct {
+	columnInfos []*tipb.ColumnInfo
+	fieldTps    []*types.FieldType
+	sc          *variable.StatementContext
+}
+
+func (e *evalContext) setColumnInfo(cols []*tipb.ColumnInfo) {
+	e.columnInfos = make([]*tipb.ColumnInfo, len(cols))
+	copy(e.columnInfos, cols)
+
+	e.fieldTps = make([]*types.FieldType, 0, len(e.columnInfos))
+	for _, col := range e.columnInfos {
+		ft := distsql.FieldTypeFromPBColumn(col)
+		e.fieldTps = append(e.fieldTps, ft)
+	}
+}
+
+// decodeColumnVals decodes data to Datum slice according to the row information.
+func (e *evalContext) decodeColumnVals(colIDs map[int64]int, handle int64, value [][]byte, row []types.Datum) error {
```

Comment:
I think we can move colIDs to evalContext to simplify the code. Because everytime `colIDs` is the same value.